### PR TITLE
Add depreciation warning for older macOS version

### PR DESCRIPTION
**Pull request by praveenkumar**
Try to test this PR on older mac and it is going to show depreciation warning.

### CODE REVIEW

**Comment by anjannath** [comment]
Do we have a lot of macOS 12 user's still who haven't updated? should we add this warning in the installer itself, in the past we directly removed support for older version of macOS in the pkg installer

**Comment by praveenkumar** [comment]
> Do we have a lot of macOS 12 user's still who haven't updated? should we add this warning in the installer itself, in the past we directly removed support for older version of macOS in the pkg installer

We don't have and having warning in installer also a good way to let user know. @cfergeau what do you think?

File: pkg/drivers/vfkit/driver_darwin.go [hunk 1]
```diff
@@ -488,36 +486,10 @@ func (d *Driver) sendSignal(s syscall.Signal) error {
...
 
-var (
-	once              sync.Once
-	macCurrentVersion string
-	errGettingVersion error
-)
-
 func (d *Driver) supportsVirtiofs() bool {
-	supportsVirtioFS, err := macosAtLeast("12.0.0")
+	supportsVirtioFS, err := darwin.AtLeast("12.0.0")
```
Comment (cfergeau) [inline pkg/drivers/vfkit/driver_darwin.go:490]: Not asking for changes, but one could argue this should be `darwin.AtLeast("21.0.0")`, and that `AtLeast` needs to be rewritten (darwin and macos versions are different things https://en.wikipedia.org/wiki/Darwin_(operating_system) )

File: pkg/crc/preflight/preflight_checks_darwin.go [hunk 2]
```diff
@@ -224,3 +225,14 @@ func fixPlistFileExists(agentConfig launchd.AgentConfig) error {
...
 	return waitForDaemonRunning()
 }
+
+func deprecationNotice() error {
+	supports, err := darwin.AtLeast("13.0.0")
+	if err != nil {
+		return err
+	}
+	if !supports {
+		logging.Warnf("This version of macOS is going to be unsupported for CRC, Please update to latest version of macOS")
```
Comment (cfergeau) [inline pkg/crc/preflight/preflight_checks_darwin.go]: I would say "Please update to macOS 13 or newer"

**Review by cfergeau — COMMENTED** [review]
looks good to me.

File: pkg/crc/preflight/preflight_checks_darwin.go (hunk 2 above)
Comment (praveenkumar) [reply pkg/crc/preflight/preflight_checks_darwin.go]: Done.

**Comment by praveenkumar** [comment]
>  should we add this warning in the installer itself

@anjannath Added, can you verify it once?

**Comment by cfergeau** [comment]
I  see `make check` failures in github actions, are they fixed in the latest iteration?

**Comment by anjannath** [comment]
> > should we add this warning in the installer itself
> 
> @anjannath Added, can you verify it once?

tested on macOS 12, showing deprecation message while installer is started..
image

**Comment by praveenkumar** [comment]
> I see make check failures in github actions, are they fixed in the latest iteration?

@cfergeau I forgot to rebase, will do that.

**Review by anjannath — APPROVED** [review]

**Pull request merged**